### PR TITLE
power: unify lock sates and move pm_auto_update to outer dir

### PR DESCRIPTION
**Pull request by zhuyanlinzyl**

### CODE REVIEW

File: drivers/power/pm_register.c [hunk 1]
```diff
@@ -56,20 +56,24 @@
...
 {
-  int ret;
-
   DEBUGASSERT(callbacks);
 
   /* Add the new entry to the end of the list of registered callbacks */
 
-  ret = pm_lock();
-  if (ret == OK)
+  if (OSINIT_OS_READY())
```
Comment (pkarashchenko) [inline drivers/power/pm_register.c:63]: I think this completely contradicts to https://github.com/apache/incubator-nuttx/pull/5849
@GUIDINGLI could you please review this change?

File: drivers/power/pm_lock.c [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+{
+  irqstate_t flags = 0;
+
+  if (up_interrupt_context() || sched_idletask())
+    {
+      flags = enter_critical_section();
+    }
+  else
+    {
+      nxsem_wait(&g_pmglobals.regsem);
```
[the remark this answers is not in the document]
Comment (pkarashchenko) [reply drivers/power/pm_lock.c:51]: yes, but the current implementation does not take into account interruption by a signal. Maybe we should use uninterrupted version or have `int pm_lock(irqstate_t *state);`

[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply drivers/power/pm_lock.c:51]: I think so, @zhuyanlinzyl

**Comment by zhuyanlinzyl** [comment]
Yes, I have update the patch, please review again @xiaoxiang781216  @pkarashchenko

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**